### PR TITLE
MNT Remove TODO comments

### PR DESCRIPTION
**Pull request by sabina-talipova**
## Parent issue
- https://github.com/silverstripe/.github/issues/116

### CODE REVIEW

File: src/Control/Controller.php [hunk 1]
```diff
@@ -146,8 +146,6 @@ public function setRequest(HTTPRequest $request): static
     /**
      * A bootstrap for the handleRequest method
      *
-     * @todo setDataModel and setRequest are redundantly called in parent::handleRequest() - sort this out
-     *
```
Comment (sabina-talipova) [inline src/Control/Controller.php:150]: - https://github.com/silverstripe/silverstripe-framework/issues/10981

File: src/Control/CookieJar.php [hunk 2]
```diff
@@ -11,10 +11,6 @@
  * This backend allows one to better test Cookie setting and separate cookie
  * handling from the core
  *
- * @todo Create a config array for defaults (eg: httpOnly, secure, path, domain, expiry)
- * @todo A getter for cookies that haven't been sent to the browser yet
- * @todo Tests / a way to set the state without hacking with $_COOKIE
- * @todo Store the meta information around cookie setting (path, domain, secure, etc)
```
Comment (sabina-talipova) [inline src/Control/CookieJar.php:17]: - https://github.com/silverstripe/silverstripe-framework/issues/10982

File: src/Control/SimpleResourceURLGenerator.php [hunk 3]
```diff
@@ -140,7 +140,6 @@ protected function resolveModuleResource(ModuleResource $resource)
             // All resources mapped directly to _resources/
             $relativePath = Path::join(RESOURCES_DIR, $relativePath);
         } elseif (stripos($relativePath ?? '', ManifestFileFinder::VENDOR_DIR . DIRECTORY_SEPARATOR) === 0) {
-            // @todo Non-public dir support will be removed in 5.0, so remove this block there
```
Comment (sabina-talipova) [inline src/Control/SimpleResourceURLGenerator.php:143]: - https://github.com/silverstripe/silverstripe-framework/issues/10983

File: src/Control/RSS/RSSFeed.php [hunk 4]
```diff
@@ -215,8 +214,6 @@ public function Description()
     /**
      * Output the feed to the browser.
      *
-     * TODO: Pass $response object to ->outputToBrowser() to loosen dependence on global state for easier testing/prototyping so dev can inject custom HTTPResponse instance.
-     *
```
Comment (sabina-talipova) [inline src/Control/RSS/RSSFeed.php:219]: - https://github.com/silverstripe/silverstripe-framework/issues/10984

File: src/Core/ClassInfo.php [hunk 5]
```diff
@@ -82,8 +82,6 @@ public static function exists($class)
     }
 
     /**
-     * @todo Move this to SS_Database or DB
-     *
```
Comment (sabina-talipova) [inline src/Core/ClassInfo.php:86]: - https://github.com/silverstripe/silverstripe-framework/issues/10985

File: src/Core/ClassInfo.php [hunk 6]
```diff
@@ -130,8 +128,6 @@ public static function getValidSubClasses($class = SiteTree::class, $includeUnba
      * Returns an array of the current class and all its ancestors and children
      * which require a DB table.
      *
-     * @todo Move this into {@see DataObjectSchema}
-     *
```
Comment (sabina-talipova) [inline src/Core/ClassInfo.php:134]: https://github.com/silverstripe/silverstripe-framework/issues/10986

File: src/Core/Extensible.php [hunk 7]
```diff
@@ -217,8 +217,6 @@ public static function add_extension($classOrExtension, $extension = null)
      * Clears any previously created singletons through {@link singleton()}
      * to avoid side-effects from stale extension information.
      *
-     * @todo Add support for removing extensions with parameters
-     *
```
Comment (sabina-talipova) [inline src/Core/Extensible.php:221]: - https://github.com/silverstripe/silverstripe-framework/issues/10987

File: src/Core/Injector/Injector.php [hunk 8]
```diff
@@ -666,10 +666,6 @@ protected function instantiate($spec, $id = null, $type = null)
     /**
      * Inject $object with available objects from the service cache
      *
-     * @todo Track all the existing objects that have had a service bound
-     * into them, so we can update that binding at a later point if needbe (ie
-     * if the managed service changes)
-     *
```
Comment (sabina-talipova) [inline src/Core/Injector/Injector.php:672]: - https://github.com/silverstripe/silverstripe-framework/issues/10988

File: src/Dev/BulkLoader_Result.php [hunk 9]
```diff
@@ -11,8 +11,6 @@
  * Encapsulates the result of a {@link BulkLoader} import
  * (usually through the {@link BulkLoader->processAll()} method).
  *
- * @todo Refactor to support lazy-loaded DataObjectSets once they are implemented.
- *
```
Comment (sabina-talipova) [inline src/Dev/BulkLoader_Result.php:15]: - https://github.com/silverstripe/silverstripe-framework/issues/10989

File: src/Dev/Constraint/ViewableDataContains.php [hunk 10]
```diff
@@ -74,8 +74,6 @@ public function evaluate($other, $description = '', $returnResult = false): ?boo
 
     /**
      * Returns a string representation of the object.
-     *
-     * @todo: add representation for more than one match
```
Comment (sabina-talipova) [inline src/Dev/Constraint/ViewableDataContains.php:78]: - https://github.com/silverstripe/silverstripe-framework/issues/10990

File: src/Core/Convert.php [hunk 11]
```diff
@@ -204,7 +204,6 @@ public static function symbol2sql($identifier, $separator = '.')
      * Warning: Does not decode array keys
      *
      * @uses html2raw()
-     * @todo Currently &#xxx; entries are stripped; they should be converted
```
Comment (sabina-talipova) [inline src/Core/Convert.php:207]: - https://github.com/silverstripe/silverstripe-framework/issues/10991

File: src/Forms/CurrencyField.php [hunk 12]
```diff
@@ -9,8 +9,6 @@
  * Limited to US-centric formats, including a hardcoded currency
  * symbol and decimal separators.
  * See {@link MoneyField} for a more flexible implementation.
- *
- * @todo Add localization support, see http://open.silverstripe.com/ticket/2931
```
Comment (sabina-talipova) [inline src/Forms/CurrencyField.php:13]: - https://github.com/silverstripe/silverstripe-framework/issues/10992

File: src/Dev/BulkLoader.php [hunk 13]
```diff
@@ -12,10 +12,6 @@
  *
  * You can configure column-handling,
  *
- * @todo Add support for adding/editing has_many relations.
- * @todo Add support for deep chaining of relation properties (e.g. Player.Team.Stats.GoalCount)
- * @todo Character conversion
- *
```
Comment (sabina-talipova) [inline src/Dev/BulkLoader.php:18]: - https://github.com/silverstripe/silverstripe-framework/issues/10993

File: src/Dev/CsvBulkLoader.php
Comment (sabina-talipova) [inline src/Dev/CsvBulkLoader.php:1]: - https://github.com/silverstripe/silverstripe-framework/issues/10994

File: src/Dev/CsvBulkLoader.php [hunk 14]
```diff
@@ -160,9 +157,6 @@ protected function getNormalisedColumnMap()
     }
 
     /**
-     * @todo Better messages for relation checks and duplicate detection
-     * Note that columnMap isn't used.
-     *
```
Comment (sabina-talipova) [inline src/Dev/CsvBulkLoader.php:165]: https://github.com/silverstripe/silverstripe-framework/issues/10995

File: src/Dev/CsvBulkLoader.php [hunk 15]
```diff
@@ -289,8 +282,6 @@ protected function processRecord($record, $columnMap, &$results, $preview = fals
      * Find an existing objects based on one or more uniqueness columns
      * specified via {@link self::$duplicateChecks}.
      *
-     * @todo support $columnMap
-     *
```
Comment (sabina-talipova) [inline src/Dev/CsvBulkLoader.php:293]: - https://github.com/silverstripe/silverstripe-framework/issues/10995

File: src/Dev/CsvBulkLoader.php (hunk 14 above)
Comment (sabina-talipova) [inline src/Dev/CsvBulkLoader.php:165]: - https://github.com/silverstripe/silverstripe-framework/issues/10996

File: src/Dev/Debug.php [hunk 16]
```diff
@@ -25,9 +25,6 @@
  * Errors handled by this class are passed along to {@link SS_Log}.
  * For configuration information, see the {@link SS_Log}
  * class documentation.
- *
- * @todo add support for user defined config: Debug::die_on_notice(true | false)
- * @todo better way of figuring out the error context to display in highlighted source
  */
```
Comment (sabina-talipova) [inline src/Dev/Debug.php:28]: - https://github.com/silverstripe/silverstripe-framework/issues/10997

File: src/Dev/Debug.php [hunk 17]
```diff
@@ -212,7 +209,6 @@ public static function require_developer_login()
             // being called again.
             // This basically calls Permission::checkMember($_SESSION['loggedInAs'], 'ADMIN');
 
-            // @TODO - Rewrite safely using DataList::filter
```
Comment (sabina-talipova) [inline src/Dev/Debug.php:215]: - https://github.com/silverstripe/silverstripe-framework/issues/10998

File: src/Dev/DevelopmentAdmin.php [hunk 18]
```diff
@@ -19,11 +19,6 @@
  *
  * Configured in framework/_config/dev.yml, with the config key registeredControllers being
  * used to generate the list of links for /dev.
- *
- * @todo documentation for how to add new unit tests and tasks
- * @todo do we need buildDefaults and generatesecuretoken? if so, register in the list
- * @todo cleanup errors() it's not even an allowed action, so can go
- * @todo cleanup index() html building
```
Comment (sabina-talipova) [inline src/Dev/DevelopmentAdmin.php:26]: - https://github.com/silverstripe/silverstripe-framework/issues/10999

File: src/Forms/CheckboxSetField.php [hunk 19]
```diff
@@ -36,21 +36,13 @@
  *   the database records.
  * - If the field name matches a database field, a comma-separated list of values will be saved to that field.  The
  *   keys can be text or numbers.
- *
- * @todo Document the different source data that can be used
- * with this form field - e.g ComponentSet, ArrayList,
- * array. Is it also appropriate to accept so many different
- * types of data when just using an array would be appropriate?
```
Comment (sabina-talipova) [inline src/Forms/CheckboxSetField.php:43]: - https://github.com/silverstripe/silverstripe-framework/issues/11000

File: src/Forms/FieldList.php [hunk 20]
```diff
@@ -411,8 +411,6 @@ public function findTab($tabName)
     /**
      * Returns the specified tab object, creating it if necessary.
      *
-     * @todo Support recursive creation of TabSets
-     *
```
Comment (sabina-talipova) [inline src/Forms/FieldList.php:415]: - https://github.com/silverstripe/silverstripe-framework/issues/11001

File: src/Forms/FieldList.php [hunk 21]
```diff
@@ -462,8 +460,6 @@ public function findOrMakeTab($tabName, $title = null)
      * Returns a named field.
      * You can use dot syntax to get fields from child composite fields
      *
-     * @todo Implement similarly to dataFieldByName() to support nested sets - or merge with dataFields()
-     *
```
Comment (sabina-talipova) [inline src/Forms/FieldList.php:466]: - https://github.com/silverstripe/silverstripe-framework/issues/11002

File: src/Forms/TimeField.php [hunk 22]
```diff
@@ -14,8 +14,6 @@
  * # Localization
  *
  * See {@link DateField}
- *
- * @todo Timezone support
```
Comment (sabina-talipova) [inline src/Forms/TimeField.php:18]: - https://github.com/silverstripe/silverstripe-framework/issues/11003

File: src/ORM/DataObjectSchema.php [hunk 23]
```diff
@@ -1153,7 +1153,7 @@ protected function checkManyManyFieldClass($parentClass, $component, $joinClass,
             if ($key === 'from') {
                 return $relationClass;
             }
-            // @todo support polymorphic 'to'
```
Comment (sabina-talipova) [inline src/ORM/DataObjectSchema.php:1156]: - https://github.com/silverstripe/silverstripe-framework/issues/11004

File: src/Forms/GroupedDropdownField.php [hunk 24]
```diff
@@ -55,8 +55,6 @@
 class GroupedDropdownField extends DropdownField
 {
 
-    // TODO remove this when GroupedDropdownField is implemented
-    // This should be one of SCHEMA_DATA_TYPE_* constants instead
```
Comment (sabina-talipova) [inline src/Forms/GroupedDropdownField.php:59]: - https://github.com/silverstripe/silverstripe-framework/issues/11005

File: src/Forms/MoneyField.php [hunk 25]
```diff
@@ -17,7 +17,6 @@
 class MoneyField extends FormField
 {
 
-    // TODO replace with `FormField::SCHEMA_DATA_TYPE_TEXT` when MoneyField is implemented
```
Comment (sabina-talipova) [inline src/Forms/MoneyField.php:20]: - https://github.com/silverstripe/silverstripe-framework/issues/11006

File: src/ORM/DB.php [hunk 26]
```diff
@@ -471,8 +471,6 @@ public static function prepared_query($sql, $parameters, $errorLevel = E_USER_ER
      * That's a limitation of the system that's due to it being written for {@link DataObject::write()},
      * which needs to do a single write on a number of different tables.
      *
-     * @todo Update this to support paramaterised queries
-     *
```
Comment (sabina-talipova) [inline src/ORM/DB.php:475]: - https://github.com/silverstripe/silverstripe-framework/issues/11007

File: src/ORM/FieldType/DBField.php [hunk 27]
```diff
@@ -579,12 +577,6 @@ public function scaffoldSearchField($title = null)
     }
 
     /**
-     * @todo documentation
-     *
-     * @todo figure out how we pass configuration parameters to
-     *       search filters (note: parameter hack now in place to pass in the required full path - using $this->name
-     *       won't work)
-     *
```
Comment (sabina-talipova) [inline src/ORM/FieldType/DBField.php:587]: - https://github.com/silverstripe/silverstripe-framework/issues/11008

File: src/ORM/FieldType/DBPolymorphicForeignKey.php [hunk 28]
```diff
@@ -20,7 +20,6 @@ public function scaffoldFormField($title = null, $params = null)
     {
         // Opt-out of form field generation - Scaffolding should be performed on
         // the has_many end, or set programmatically.
-        // @todo - Investigate suitable FormField
```
Comment (sabina-talipova) [inline src/ORM/FieldType/DBPolymorphicForeignKey.php:23]: - https://github.com/silverstripe/silverstripe-framework/issues/11009

File: tests/php/ORM/DataObjectTest.php [hunk 29]
```diff
@@ -1633,12 +1624,6 @@ public function testMerge()
             null,
             'merge() with $overwriteWithEmpty overwrites empty fields on left object'
         );
-
-        // @todo test "left" priority flag
-        // @todo test includeRelations flag
-        // @todo test includeRelations in combination with overwriteWithEmpty
-        // @todo test has_one relations
-        // @todo test has_many and many_many relations
```
Comment (sabina-talipova) [inline tests/php/ORM/DataObjectTest.php:1641]: - https://github.com/silverstripe/silverstripe-framework/issues/11010

File: tests/php/Forms/RequiredFieldsTest.php [hunk 30]
```diff
@@ -5,9 +5,6 @@
 use SilverStripe\Dev\SapphireTest;
 use SilverStripe\Forms\RequiredFields;
 
-/**
- * @todo Test the validation method php()
- */
```
Comment (sabina-talipova) [inline tests/php/Forms/RequiredFieldsTest.php:10]: - https://github.com/silverstripe/silverstripe-framework/issues/11011

File: tests/php/Forms/FieldListTest.php
Comment (sabina-talipova) [inline tests/php/Forms/FieldListTest.php:1]: - https://github.com/silverstripe/silverstripe-framework/issues/11012

File: tests/php/Forms/CurrencyFieldReadonlyTest.php [hunk 31]
```diff
@@ -37,9 +37,6 @@ public function testFieldWithOutValue()
         $this->assertStringContainsString('AUD0.00', $result, 'The value should be rendered');
     }
 
-    /**
-     * @todo: Update the expectation when intl for currencies is implemented
-     */
```
Comment (sabina-talipova) [inline tests/php/Forms/CurrencyFieldReadonlyTest.php:42]: - https://github.com/silverstripe/silverstripe-framework/issues/11013

File: tests/php/Forms/CurrencyFieldDisabledTest.php [hunk 32]
```diff
@@ -18,9 +18,6 @@ public function testFieldWithValue()
         $this->assertStringContainsString('$5.00', $result, 'The value should be rendered');
     }
 
-    /**
-     * @todo: Update the expectation when intl for currencies is implemented
-     */
```
Comment (sabina-talipova) [inline tests/php/Forms/CurrencyFieldDisabledTest.php:23]: - https://github.com/silverstripe/silverstripe-framework/issues/11013

File: tests/php/Core/PhpSyntaxTest.php [hunk 33]
```diff
@@ -18,7 +18,6 @@ protected function setUp(): void
     public function testShortTagsOffWillWork()
     {
         // Ignore this test completely if running the test suite on windows
-        // TODO: Make it work on all platforms, by building an alternative to find | grep.
```
Comment (sabina-talipova) [inline tests/php/Core/PhpSyntaxTest.php:21]: - https://github.com/silverstripe/silverstripe-framework/issues/11014

File: tests/php/Core/ObjectTest.php
Comment (sabina-talipova) [inline tests/php/Core/ObjectTest.php:1]: - https://github.com/silverstripe/silverstripe-framework/issues/11015

File: src/Security/Member.php [hunk 34]
```diff
@@ -114,7 +114,7 @@ class Member extends DataObject
      * statement.
      *
      * @var array
-     * @todo Generic implementation of $searchable_fields on DataObject,
```
Comment (sabina-talipova) [inline src/Security/Member.php:117]: - https://github.com/silverstripe/silverstripe-framework/issues/11016

File: src/Security/Member.php [hunk 35]
```diff
@@ -446,7 +446,6 @@ public function isPasswordExpired(): bool
      */
     public function beforeMemberLoggedIn()
     {
-        // @todo Move to middleware on the AuthenticationMiddleware IdentityStore
```
Comment (sabina-talipova) [inline src/Security/Member.php:449]: - https://github.com/silverstripe/silverstripe-framework/issues/11016

File: src/ORM/Filters/WithinRangeFilter.php [hunk 36]
```diff
@@ -4,11 +4,6 @@
 
 use SilverStripe\ORM\DataQuery;
 
-/**
- * Incomplete.
- *
- * @todo add to tests
- */
```
Comment (sabina-talipova) [inline src/ORM/Filters/WithinRangeFilter.php:11]: - https://github.com/silverstripe/silverstripe-framework/issues/11017

File: src/ORM/Filters/FulltextFilter.php [hunk 37]
```diff
@@ -25,7 +25,6 @@
  *   ];
  * </code>
  *
- * @todo Add support for databases besides MySQL
```
Comment (sabina-talipova) [inline src/ORM/Filters/FulltextFilter.php:28]: - https://github.com/silverstripe/silverstripe-framework/issues/11018

File: src/ORM/FieldType/DBPrimaryKey.php [hunk 38]
```diff
@@ -7,8 +7,6 @@
 
 /**
  * A special type Int field used for primary keys.
- *
- * @todo Allow for custom limiting/filtering of scaffoldFormField dropdown
```
Comment (sabina-talipova) [inline src/ORM/FieldType/DBPrimaryKey.php:11]: - https://github.com/silverstripe/silverstripe-framework/issues/11019

File: src/ORM/FieldType/DBDatetime.php [hunk 39]
```diff
@@ -30,7 +30,6 @@
  * );
  * </code>
  *
- * @todo Add localization support, see http://open.silverstripe.com/ticket/2931
```
Comment (sabina-talipova) [inline src/ORM/FieldType/DBDatetime.php:33]: - https://github.com/silverstripe/silverstripe-framework/issues/11020

File: src/Forms/FormField.php [hunk 40]
```diff
@@ -1447,8 +1443,6 @@ public function getSchemaData()
     }
 
     /**
-     * @todo Throw exception if value is missing, once a form field schema is mandatory across the CMS
-     *
```
Comment (sabina-talipova) [inline src/Forms/FormField.php:1451]: - https://github.com/silverstripe/silverstripe-framework/issues/11021

File: src/Forms/FormField.php [hunk 41]
```diff
@@ -1425,8 +1423,6 @@ public function getSchemaComponent()
      *
      * @param array $schemaData - The data to be merged with $this->schemaData.
      * @return FormField
-     *
-     * @todo Add deep merging of arrays like `data` and `attributes`.
```
Comment (sabina-talipova) [inline src/Forms/FormField.php:1429]: - https://github.com/silverstripe/silverstripe-framework/issues/11022

File: src/ORM/FieldType/DBComposite.php [hunk 42]
```diff
@@ -12,8 +12,6 @@
  * Extend this class when designing a {@link DBField} that doesn't have a 1-1 mapping with a database field.
  * This includes multi-value fields and transformed fields
  *
- * @todo Unittests for loading and saving composite values (see GIS module for existing similar unittests)
- *
```
Comment (sabina-talipova) [inline src/ORM/FieldType/DBComposite.php:16]: - https://github.com/silverstripe/silverstripe-framework/issues/11023

File: src/ORM/DataObject.php [hunk 43]
```diff
@@ -99,9 +99,6 @@
  * the results are cached in memory through {@link cachedCall()}.
  *
  *
- * @todo Add instance specific removeExtension() which undos loadExtraStatics()
- *  and defineMethods()
- *
```
Comment (sabina-talipova) [inline src/ORM/DataObject.php:104]: - https://github.com/silverstripe/silverstripe-framework/issues/11024

File: src/ORM/DataObject.php [hunk 44]
```diff
@@ -1151,7 +1144,6 @@ public function merge($rightObj, $priority = 'right', $includeRelations = true,
                 continue;
             }
 
-            // TODO remove redundant merge of has_one fields
```
Comment (sabina-talipova) [inline src/ORM/DataObject.php:1154]: - https://github.com/silverstripe/silverstripe-framework/issues/11025

File: src/ORM/DataObject.php [hunk 45]
```diff
@@ -1752,10 +1744,6 @@ public function delete()
             throw new LogicException("DataObject::delete() called on a DataObject without an ID");
         }
 
-        // TODO: This is quite ugly.  To improve:
-        //  - move the details of the delete code in the DataQuery system
-        //  - update the code to just delete the base table, and rely on cascading deletes in the DB to do the rest
-        //    obviously, that means getting requireTable() to configure cascading deletes ;-)
```
Comment (sabina-talipova) [inline src/ORM/DataObject.php:1758]: - https://github.com/silverstripe/silverstripe-framework/issues/11026

File: src/ORM/DataObject.php
Comment (sabina-talipova) [inline src/ORM/DataObject.php:1]: - https://github.com/silverstripe/silverstripe-framework/issues/11027

File: src/ORM/Connect/DBSchemaManager.php [hunk 46]
```diff
@@ -350,8 +350,6 @@ protected function transInitTable($table)
      * Generate the following table in the database, modifying whatever already exists
      * as necessary.
      *
-     * @todo Change detection for CREATE TABLE $options other than "Engine"
-     *
```
Comment (sabina-talipova) [inline src/ORM/Connect/DBSchemaManager.php:354]: - https://github.com/silverstripe/silverstripe-framework/issues/11028

File: src/ORM/Connect/DBSchemaManager.php [hunk 47]
```diff
@@ -974,7 +970,6 @@ abstract public function dropDatabase($name);
      * @param string $tableName The name of the table.
      * @param string $indexName The name of the index.
      * @param array $indexSpec The specification of the index, see Database::requireIndex() for more details.
-     * @todo Find out where this is called from - Is it even used? Aren't indexes always dropped and re-added?
```
Comment (sabina-talipova) [inline src/ORM/Connect/DBSchemaManager.php:977]: - https://github.com/silverstripe/silverstripe-framework/issues/11029

File: src/Forms/GridField/GridField.php [hunk 48]
```diff
@@ -342,8 +342,6 @@ public function getComponents()
     /**
      * Cast an arbitrary value with the help of a $castingDefinition.
      *
-     * @todo refactor this into GridFieldComponent
-     *
```
Comment (sabina-talipova) [inline src/Forms/GridField/GridField.php:346]: - https://github.com/silverstripe/silverstripe-framework/issues/11030

File: sake
Comment (sabina-talipova) [inline sake:1]: - https://github.com/silverstripe/silverstripe-framework/issues/11031

File: src/Control/HTTP.php [hunk 49]
```diff
@@ -121,7 +121,7 @@ public static function urlRewriter($content, $code)
             $regExps[] = "/(<{$tagPrefix}[^>]*$attrib *= *)([^\"' ]*)( )/i";
         }
         // Replace css styles
-        // @todo - http://www.css3.info/preview/multiple-backgrounds/
+        // @ - http://www.css3.info/preview/multiple-backgrounds/
```
Comment (GuySartorelli) [inline src/Control/HTTP.php]:
```suggestion
```

File: src/ORM/FieldType/DBPolymorphicForeignKey.php [hunk 50]
```diff
@@ -20,7 +20,6 @@ public function scaffoldFormField($title = null, $params = null)
     {
         // Opt-out of form field generation - Scaffolding should be performed on
         // the has_many end, or set programmatically.
```
Comment (GuySartorelli) [inline src/ORM/FieldType/DBPolymorphicForeignKey.php]:
```suggestion
        // Don't provide scaffolded form field generation - Scaffolding should be performed on
        // the has_many end, or set programmatically.
```

File: src/ORM/Queries/SQLAssignmentRow.php [hunk 51]
```diff
@@ -68,7 +68,7 @@ protected function parseAssignment($value)
                     $parameters = [$parameters];
                 }
 
-                // @todo Some input sanitisation checking the key contains the
+                // Some input sanitisation checking the key contains the
                 // correct number of ? placeholders as the number of parameters
```
Comment (GuySartorelli) [inline src/ORM/Queries/SQLAssignmentRow.php]:
```suggestion
```
Please always remove the full todo comment.

File: src/Security/Member.php [hunk 52]
```diff
@@ -114,7 +114,7 @@ class Member extends DataObject
      * statement.
      *
      * @var array
-     * @todo Generic implementation of $searchable_fields on DataObject,
+     * Generic implementation of $searchable_fields on DataObject,
      * with definition for different searching algorithms
      * (LIKE, FULLTEXT) and default FormFields to construct a searchform.
```
Comment (GuySartorelli) [inline src/Security/Member.php]:
```suggestion
```
Please always remove the _full_ todo comment.

File: src/Security/MemberAuthenticator/MemberLoginForm.php [hunk 53]
```diff
@@ -132,9 +131,6 @@ protected function getFormFields()
         $label = Member::singleton()->fieldLabel(Member::config()->get('unique_identifier_field'));
         $fields = FieldList::create(
             HiddenField::create("AuthenticationMethod", null, $this->getAuthenticatorClass(), $this),
-            // Regardless of what the unique identifier field is (usually 'Email'), it will be held in the
-            // 'Email' value, below:
```
Comment (GuySartorelli) [inline src/Security/MemberAuthenticator/MemberLoginForm.php]: These were not part of the todo comment - they're adding useful context that this isn't _necessarily_ actually an email address. Please bring this comment back.

File: src/Security/Permission.php [hunk 54]
```diff
@@ -133,7 +133,7 @@ public static function check($code, $arg = "any", $member = null, $strict = true
 
     /**
      * Flush the permission cache, for example if you have edited group membership or a permission record.
-     * @todo Call this whenever Group_Members is added to or removed from
+     * Call this whenever Group_Members is added to or removed from
```
Comment (GuySartorelli) [inline src/Security/Permission.php]:
```suggestion
```
That was part of the todo comment.

File: tests/behat/src/CmsUiContext.php [hunk 55]
```diff
@@ -594,7 +594,6 @@ public function thePreviewDoesNotContain($content)
      */
     public function clickLinkInPreview($link)
     {
-        // TODO Remove once we have native support in Mink and php-webdriver,
         // see https://groups.google.com/forum/#!topic/behat/QNhOuGHKEWI
```
Comment (GuySartorelli) [inline tests/behat/src/CmsUiContext.php]:
```suggestion
```
That's part of the todo comment

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Control/HTTP.php (hunk 49 above)
Comment (sabina-talipova) [reply src/Control/HTTP.php]: Updated

File: src/ORM/FieldType/DBPolymorphicForeignKey.php (hunk 50 above)
Comment (sabina-talipova) [reply src/ORM/FieldType/DBPolymorphicForeignKey.php]: Updated

File: src/ORM/Queries/SQLAssignmentRow.php (hunk 51 above)
Comment (sabina-talipova) [reply src/ORM/Queries/SQLAssignmentRow.php]: Updated

File: src/Security/Member.php (hunk 52 above)
Comment (sabina-talipova) [reply src/Security/Member.php]: Updated

File: src/Security/MemberAuthenticator/MemberLoginForm.php (hunk 53 above)
Comment (sabina-talipova) [reply src/Security/MemberAuthenticator/MemberLoginForm.php]: Updated

File: src/Security/Permission.php (hunk 54 above)
Comment (sabina-talipova) [reply src/Security/Permission.php]: Updated

File: tests/behat/src/CmsUiContext.php (hunk 55 above)
Comment (sabina-talipova) [reply tests/behat/src/CmsUiContext.php]: Updated

File: src/Security/MemberAuthenticator/MemberLoginForm.php
Comment (GuySartorelli) [inline src/Security/MemberAuthenticator/MemberLoginForm.php:1]: There's still a todo comment in line 87
> // @todo find a more elegant way to handle this

**Review by GuySartorelli — CHANGES_REQUESTED** [review]

File: src/Security/MemberAuthenticator/MemberLoginForm.php
Comment (sabina-talipova) [reply src/Security/MemberAuthenticator/MemberLoginForm.php:1]: Updated

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**